cleanup-passes/v3-elements.ts

```diff
@@ -0,0 +1,172 @@
+/**
+ * @license
+ * Copyright (c) 2018 The Polymer Project Authors. All rights reserved.  This
+ * code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt The complete set of authors may be found
+ * at http://polymer.github.io/AUTHORS.txt The complete set of contributors may
+ * be found at http://polymer.github.io/CONTRIBUTORS.txt Code distributed by
+ * Google as part of the polymer project is also subject to an additional IP
+ * rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+'use strict';
+
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import * as yaml from 'js-yaml';
+import {execFile} from 'child_process';
+import {promisify} from 'util';
+
+import {register} from '../cleanup-pass';
+import {ElementRepo} from '../element-repo';
+import {makeCommit} from './util';
+
+const execFilePromise = promisify(execFile);
+
+const filesWeMightChange = new Set<string>([
+  '.travis.yml',
+  '.npmignore',
+  '.gitignore',
+  'package.json',
+  'package-lock.json',
+]);
+
+/**
+ * Perform updates for 3.0 Polymer elements.
+ */
+async function v3ElementPass(element: ElementRepo): Promise<void> {
+  await updateNpmIgnore(element);
+  await updateGitIgnore(element);
+  await updateTravisYaml(element);
+  await updatePackageJson(element);
+  await regeneratePackageLock(element);
+
+  let shouldCommit = false;
+  for (const changedFile of await element.repo.getStatus()) {
+    const filePath = changedFile.path();
+    if (filesWeMightChange.has(filePath)) {
+      shouldCommit = true;
+    } else {
+      throw new Error(
+          `${element.ghRepo.name}: Unexpected changed file: ${filePath}`);
+    }
+  }
+  if (shouldCommit) {
+    await makeCommit(
+        element,
+        [...filesWeMightChange],
+        'Tedium automated v3 element updates.');
+  }
+}
+
+register({name: 'v3-elements', pass: v3ElementPass});
+
+const addToNpmIgnore = [
+  '*.tgz',
+  '.github',
+  '.travis.yml',
+  'formatconfig.json',
+  'gen-tsd.json',
+  'test/',
+  'wct.conf.json',
+];
+
+async function updateNpmIgnore(element: ElementRepo): Promise<void> {
+  const npmIgnorePath = path.join(element.dir, '.npmignore');
+  let npmIgnore = '';
+  if (await fse.pathExists(npmIgnorePath)) {
+    npmIgnore = await fse.readFile(npmIgnorePath, 'utf8');
```

Comment:
If it does not exist, I think the correct thing to do to preserve existing behavior is to copy .gitignore, but without the `.d.ts` line

That suggests that we should also run the gitignore pass first.